src/tests/plugins/ticketoutputpdf/test_api.py

```diff
@@ -57,32 +70,138 @@ def env():
-    )
+    res['id'] = env[1].pk
+    res['item_assignments'][0]['item'] = env[2].pk
+    r = token_client.get('/api/v1/organizers/{}/events/{}/ticketlayouts/{}/'.format(
+        env[0].organizer.slug, env[0].slug, env[1].pk)).data
     assert r == res
+
+
+@pytest.mark.django_db
+def test_api_create(env, token_client):
+    r = token_client.post(
+        '/api/v1/upload',
+        data={
+            'media_type': 'application/pdf',
+            'file': ContentFile('file.pdf', 'invalid pdf content')
+        },
+        format='upload',
+        HTTP_CONTENT_DISPOSITION='attachment; filename="file.pdf"',
+    )
+    assert r.status_code == 201
+    file_id_png = r.data['id']
+
+    resp = token_client.post(
+        '/api/v1/organizers/{}/events/{}/ticketlayouts/'.format(env[0].slug, env[0].slug),
+        {
+            'name': 'Foo',
+            'default': False,
+            "background": file_id_png,
+            'layout': [],
+        },
+        format='json'
+    )
+    assert resp.status_code == 201
+    tl = env[0].ticket_layouts.get(pk=resp.data["id"])
+    assert tl.background
+
+
+@pytest.mark.django_db
+def test_api_create_validate_default(env, token_client):
+    resp = token_client.post(
+        '/api/v1/organizers/{}/events/{}/ticketlayouts/'.format(env[0].slug, env[0].slug),
+        {
+            'name': 'Foo',
+            'default': True,
+            'layout': [],
+        },
+        format='json'
+    )
+    assert resp.status_code == 400
+    assert resp.data == {"non_field_errors": ["You cannot have two layouts with default = True"]}
+
+
+@pytest.mark.django_db
+def test_api_create_validate_layout(env, token_client):
+    resp = token_client.post(
+        '/api/v1/organizers/{}/events/{}/ticketlayouts/'.format(env[0].slug, env[0].slug),
+        {
+            'name': 'Foo',
+            'default': True,
+            'layout': [
+                {
+                    "foo": "bar"
+                }
+            ],
+        },
+        format='json'
+    )
+    assert resp.status_code == 400
+    assert resp.data["layout"][0].startswith("Your layout file is not a valid layout. Error message:")
+
+
+@pytest.mark.django_db
+def test_api_update(env, token_client):
+    r = token_client.post(
+        '/api/v1/upload',
+        data={
+            'media_type': 'application/pdf',
+            'file': ContentFile('file.pdf', 'invalid pdf content')
+        },
+        format='upload',
+        HTTP_CONTENT_DISPOSITION='attachment; filename="file.pdf"',
+    )
+    assert r.status_code == 201
+    file_id_png = r.data['id']
+
+    resp = token_client.patch(
+        '/api/v1/organizers/{}/events/{}/ticketlayouts/{}/'.format(env[0].slug, env[0].slug, env[1].pk),
+        {
+            "name": "Bar",
+            "background": file_id_png,
+        },
+        format='json'
+    )
+    assert resp.status_code == 200
+    env[1].refresh_from_db()
+    assert env[1].name == "Bar"
+
```

Comment:
Should the background be checked as well like in `test_api_create`? Also, should we test whether an update to the layout works?

Reply:
sure